Library/QuickPulseStateManager.ts

```diff
@@ -144,7 +146,7 @@ class QuickPulseStateManager {
         this._sender.post(envelope, this._quickPulseDone.bind(this));
     }
 
-    private _quickPulseDone(shouldPOST: boolean, res: http.IncomingMessage): void {
+    private _quickPulseDone(shouldPOST: boolean, res: http.IncomingMessage | {statusCode: boolean}): void {
```

Comment:
What does it mean for status code to be a boolean?

Reply:
Also, I suspect this may be buggy, since a status code of `false` will pass the success if below, because `false < 300` is `true` (somehow)

Reply:
Meant to make it optional so that I can stop sending to QPS when the requests send an error. [`done(false)`](https://github.com/Microsoft/ApplicationInsights-node.js/pull/504/files#diff-50b8f385a3e9074029ab82dd36ecfa8cR56)